Phalanx/Assets/Scripts/Entity/Unit.cs

```diff
@@ -7,43 +7,86 @@ namespace Olympus.Phalanx.Entity
 {
     public class Unit : MonoBehaviour, IOccupant
     {
+        //need time since last moved.
+        //Need destination Tile .
 
+        private float timeStartedMovement;
+        private float distanceToNextTile;
+        private bool canMove;
         // Use this for initialization
         void Start()
         {
-
         }
 
         // Update is called once per frame
         void Update()
         {
+            if (canMove)
+                move();
+        }
 
+        public void move()
+        {
+            //Why does is it giving me null reference? I break pointed to it and clearly is there and correct val
+            if (transform.position != destination.transform.position)
+            {
+                Debug.Log("Moving");
+                float timeSinceLastMovement = Time.time - timeStartedMovement;
+                float distanceToMovePerFrame = timeSinceLastMovement / distanceToNextTile;
+                transform.position = Vector3.Lerp(tile.transform.position, destination.transform.position, distanceToMovePerFrame);
```

Comment:
You're supposed to give the fraction of the linear interpolation that has elapsed.

For example: if the lerp is supposed to take one second then half a second into it you pass it 0.5.